### PR TITLE
Add prob plots to HLAPI

### PR DESCRIPTION
**Pull request by samvanstroud**
## Summary

This pull request introduces the following changes

* Add function to plot tagger probabiltity histograms in the high level api
* Add support for WP vlines in the tagger discriminant plots

Relates to the following issues

*
*

## Conformity
- [x] [Changelog entry](https://github.com/umami-hep/puma/blob/main/changelog.md)
- [x] [Documentation](https://umami-hep.github.io/puma/)

### CODE REVIEW

File: puma/tests/test_histogram.py [hunk 1]
```diff
@@ -390,14 +390,14 @@ def test_draw_vlines_histogram(self):
...
-            vlines_xvalues=[1, 2, 3],
+            xs=[1, 2, 3],
         )
         hist_plot.draw()
 
         plotname = "test_draw_vlines_histogram.png"
         hist_plot.savefig(f"{self.actual_plots_dir}/{plotname}")
         # Uncomment line below to update expected image
-        # hist_plot.savefig(f"{self.expected_plots_dir}/{plotname}")
+        hist_plot.savefig(f"{self.expected_plots_dir}/{plotname}")
```
Comment (dkobylianskii) [inline puma/tests/test_histogram.py:400]: I think that this line and similar lines below should be commented

File: puma/tests/expected_plots/test_draw_vlines_histogram_same_height.png
Comment (dkobylianskii) [inline puma/tests/expected_plots/test_draw_vlines_histogram_same_height.png:1]: For some reason, labels of vlines are not shown, when we have them in the code

**Review by dkobylianskii — COMMENTED** [review]
This looks very nice, thanks!
I added some comments

File: puma/tests/test_histogram.py (hunk 1 above)
Comment (samvanstroud) [reply puma/tests/test_histogram.py]: Good catch, fixed

File: puma/tests/expected_plots/test_draw_vlines_histogram_same_height.png
Comment (samvanstroud) [reply puma/tests/expected_plots/test_draw_vlines_histogram_same_height.png:1]: I changed the behaviour in the case where no labels are specified. The previous behaviour was to use the x-values `* 100` and convert to percent if no labels were specified. I have no idea why this was done and I don't think it makes sense to keep it that way.

The test for `test_draw_vlines_histogram_custom_labels` is unchanged -- when the labels are specified they are added correctly. Only the case where the labels are not specified is changed. Hope that makes sense

Comment (dkobylianskii) [reply puma/tests/expected_plots/test_draw_vlines_histogram_same_height.png:1]: I see, thanks, I was a bit confused.

**Comment by samvanstroud** [comment]
Thanks for the look @dkobylianskii, coverage should be fixed with the last two commits.

**Review by dkobylianskii — APPROVED** [review]

**Pull request merged**